src/lifecycle.mjs

```diff
@@ -128,14 +134,18 @@ export function extract(strategy, worker, messageRouter, args = []) {
       } else {
         const result = checkResult(strategy.module.update(message));
 
-        if (!result)
+        if (!result) {
+          messageRouter.off(`${strategy.module.name}-${type}`, callback);
+          clearInterval(interval);
           reject(
             `Strategy "${
               strategy.module.name
             }" and call init didn't return a valid result: "${JSON.stringify(
               result
             )}"`
           );
+          return;
+        }
```

Comment:
This is something that I believe I missed in my last PR. It could have been a cause for a future bug.